### ISSUE TITLE
alarm management templates - Form

### ISSUE THREAD

**Issue by Derryrover** [nens/lizard-management-client#446]
https://xd.adobe.com/view/113cd145-2091-4d91-45c9-198c4b05f5bd-670d/screen/c4bd39bc-f5d4-4b89-9d5f-05e4b33e76ce

**Comment by lexvand** [comment]
@hoanphungt See: nens/lizard-nxt#4821

So there are two fields that the message template should be stored to. I think it is best to read it from the `text` field and store to both.

We may have to reconsider how we want to deal with this in the future. We would have to introduce an option to store as HTML, but in that case we should also read from the `html` field. In our first implementation we used to have two separate fields in the form, but this was considered to be too difficult for the users.

If you have any brilliant ideas about a different solution for this please let me know.

**Comment by hoanphungt** [comment]
@lexvand 
I agree with your solution here. Best to store text to both fields to avoid confusion. I saw it as a front-end bug as the `POST` request included the text value for both `text` and `html` fields but the `PATCH` request only contains `text` field. So the `html` field is missing here.

**Comment by Derryrover** [comment]
@lexvand 
@hoanphungt 
Hi
I noticed this sms email template problem and observe as follows:

- In the existing situation a template cannot be changed from sms to email or visa-versa. Only on creation this can be decided: https://nxt3.staging.lizard.net/management/#/alarms/templates/28
- In the new design this seems undecided yet. I advice to keep things same as current situations: do for now not allow changing email/sms after creation. https://xd.adobe.com/view/63185198-afa0-4a27-9d61-dc0395753bf9-364b/screen/758b61b8-e98a-4133-8e92-76ef5089e0c5/
- What would be logical is that a 'sms / text' template should read from and update the 'text' field. A email template should read from and update the 'html' field. This is not what we do currently in the client. In both cases we read from + update the 'text' field in this seems the origin of the bug.
- The fix you propose above: store in both fields and read from the text field would work, but seems to defeat the purpose of having 2 fields on the backend.

kr Tom

**Comment by lexvand** [comment]
Sounds like a good solution, based on the given constraints.

**Comment by hoanphungt** [comment]
@lexvand @LirryPinter 
Following the design for the template form here: https://xd.adobe.com/view/63185198-afa0-4a27-9d61-dc0395753bf9-364b/screen/41b3523c-4db9-43c0-820d-f6d79c269136/, the parameters are now replaced by parameter blocks with represented texts (lets call them labels). Here are the list of available parameters and their labels that I am using:

1. first_name: first name recipient
2. last_name: last name recipient
3. email: email recipient
4. alarm_name: name alarm
5. warning_value: value threshold
6. warning_level: status threshold
7. threshold_timestamp: timestamp threshold
8. trigger_timestamp: trigger timestamp
9. warning_timestamp: warning timestamp
10. first_timestamp: first timestamp

Could you two have a look at the above list and see if any improvement needed for the labels?

**Comment by hoanphungt** [comment]
@Derryrover 
Hi Tom,

Could you review this PR: https://github.com/nens/lizard-management-client/pull/678? Compared to the design, there is one major change which is the SMS/EMAIL selection. I used the standard select dropdown for this field instead of creating a new selection field for it. Let me know what you think. I think it fits quite right with the rest and we can reuse the select dropdown component.

Kr, Hoan

**Comment by lexvand** [comment]
@hoanphungt I think we should keep the descriptions of the old management screen. Can that be included as a hover text for each block?

**Comment by hoanphungt** [comment]
yes it can. Currently when hover over it, you see the parameter text, for example: `[[var:first_name]]`. I can make it to show the description of the parameter instead.

**Comment by Derryrover** [comment]
@hoanphungt 

Looks good! approved.
